# HG changeset patch
# User andrew
# Date 1518051572 0
#      Thu Feb 08 00:59:32 2018 +0000
# Node ID 3ade0115344b77e6d00462044e0cf68722685bfe
# Parent  3df435e5c2807c1596d5ff98698b493c82384317
8196978, PR3517: JDK-8187667 fails on GCC 4.4.7 as found on RHEL 6
Summary: Move pragma to above function definition and turn off push/pop pragma warnings on GCC < 4.6
Reviewed-by: aph

diff --git a/src/os/linux/vm/os_linux.inline.hpp b/src/os/linux/vm/os_linux.inline.hpp
--- openjdk/hotspot/src/os/linux/vm/os_linux.inline.hpp
+++ openjdk/hotspot/src/os/linux/vm/os_linux.inline.hpp
@@ -114,12 +114,12 @@
   return ::ftruncate64(fd, length);
 }
 
+// readdir_r has been deprecated since glibc 2.24.
+// See https://sourceware.org/bugzilla/show_bug.cgi?id=19056 for more details.
+PRAGMA_DIAG_PUSH
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
 inline struct dirent* os::readdir(DIR* dirp, dirent *dbuf)
 {
-// readdir_r has been deprecated since glibc 2.24.
-// See https://sourceware.org/bugzilla/show_bug.cgi?id=19056 for more details.
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
 
   dirent* p;
   int status;
@@ -135,8 +135,8 @@
   } else
     return p;
 
-#pragma GCC diagnostic pop
 }
+PRAGMA_DIAG_POP
 
 inline int os::closedir(DIR *dirp) {
   assert(dirp != NULL, "argument is NULL");
